src/bindgen/library.rs

```diff
@@ -443,4 +450,88 @@ impl Library {
             x.mangle_paths(&monomorphs);
         }
     }
+
+    // Replace all associated types with concrete types
+    fn replace_assoc_types(&mut self) {
+        let assoc_map = &self.assoc_types;
+
+        // Replace associated types in constants
+        self.constants.for_all_items_mut(|const_| {
+            replace_concrete(&mut const_.ty, assoc_map);
+        });
+
+        // Replace associated types in 'static's
+        self.globals.for_all_items_mut(|static_| {
+            replace_concrete(&mut static_.ty, assoc_map);
+        });
+
+        // Replace associated types in enums
+        self.enums.for_all_items_mut(|enum_| {
+            for variant in enum_.variants.iter_mut() {
+                if let VariantBody::Body { body, .. } = &mut variant.body {
+                    for field in body.fields.iter_mut() {
+                        replace_concrete(&mut field.ty, assoc_map);
+                    }
+                }
+            }
+        });
+
+        // Replace associated types in structs
+        self.structs.for_all_items_mut(|struct_| {
+            for field in struct_.fields.iter_mut() {
+                replace_concrete(&mut field.ty, assoc_map);
+            }
+        });
+
+        // Replace associated types in unions
+        self.unions.for_all_items_mut(|union_| {
+            for field in union_.fields.iter_mut() {
+                replace_concrete(&mut field.ty, assoc_map);
+            }
+        });
+
+        // Replace associated types in type aliases (C typedef)
+        self.typedefs.for_all_items_mut(|typedef| {
+            replace_concrete(&mut typedef.aliased, assoc_map);
+        });
+
+        // Replace associated types in function signatures
+        for func in self.functions.iter_mut() {
+            // Check return type
+            replace_concrete(&mut func.ret, assoc_map);
+
+            // Check arguments type
+            for arg in func.args.iter_mut() {
+                replace_concrete(&mut arg.ty, assoc_map);
+            }
+        }
+    }
+}
+
+// Search and replace specific associated types with concrete types
+fn replace_concrete<'a>(ty: &mut Type, type_map: &'a HashMap<AssocTypeId, (Type, u8)>) {
```

Comment:
```suggestion
fn replace_concrete(&self, ty: &Type) -> Type {
```

please change API to this